src/oom/oomd-util.c

```diff
@@ -280,6 +280,179 @@ int oomd_cgroup_kill(const char *path, bool recurse, bool dry_run) {
         return !set_isempty(pids_killed);
 }
 
+static void oom_kill_state_free(struct OomdKillState *ks) {
+        if (ks)
+                set_remove(ks->manager->prekill_ctxs, ks);
+        free(ks);
+}
+
+DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(struct OomdKillState*, oom_kill_state_free, NULL);
+
+static struct OomdKillState* oomd_kill_state_dec(struct OomdKillState *ks) {
+        int r = oomd_cgroup_kill(ks->ctx->path, true);
+        if (r < 0)
+                log_debug_errno(r, "oomd failed to kill cgroup %s: %m", ks->ctx->path);
+        oom_kill_state_free(ks);
+        return NULL;
+}
+
+DEFINE_PRIVATE_TRIVIAL_REF_UNREF_FUNC(struct OomdKillState, oomd_kill_state, oomd_kill_state_dec);
+
+int clean_prekills(sd_event_source *e, void *userdata)
+{
+        struct Set *prekill_ctxs = userdata;
+        struct OomdKillState *ks;
+
+        assert(prekill_ctxs);
+
+        SET_FOREACH(ks, prekill_ctxs) {
+                log_debug("Cleaning up unfinished prekill hook state for cgroup %s", ks->ctx->path);
+                oom_kill_state_free(ks);
+        }
+
+        return 0;
+}
+
+static int prekill_callback(
+                sd_varlink *link,
+                sd_json_variant *parameters,
+                const char *error_id,
+                sd_varlink_reply_flags_t flags,
+                void *userdata) {
+
+        struct OomdKillState *ks = ASSERT_PTR(userdata);
+
+        if (error_id)
+                log_warning("oomd prekill hook returned error: %s", error_id);
+        else
+                log_info("oomd prekill hook finished for cgroup %s", ks->ctx->path);
+
+        sd_varlink_unref(link);
+        oomd_kill_state_unref(ks);
+
```

Comment:
double empty line

Reply:
You mean double it, or it was doubled?

Reply:
heh, English. I meant there are two empty lines

Reply:
I see only one between oomd_kill_state_unref and return, perhaps it was for an outdated push?